### PR TITLE
C51-286: Auto populate contact

### PR DESCRIPTION
**Pull request by reneolivo**
## Overview
This PR automatically selects the current contact when creating a new case from the contact's case tab.

## Before
pr-before

## After
pr-after

## Technical details

The solution was simple: to pass the `cid` (contact id) parameter to the form URL:

```html
<a ng-href="{{ 'civicrm/case/add' | civicaseCrmUrl:{cid: contactId, reset: 1, action: 'add', context: 'standalone'} }}">
```

The contact ID is provided by ` Contact.getContactIDFromUrl()`:

```js
$scope.contactId = Contact.getContactIDFromUrl();
```

Since this value is constant all other instances of `Contact.getContactIDFromUrl()` were replaced by `$scope.contactId`.

### CODE REVIEW

File: ang/test/civicase/ContactCaseTab.spec.js [hunk 1]
```diff
@@ -1 +1,40 @@
-// Todo write unit tests for ContactCaseTab.js
+/* eslint-env jasmine */
+
+(function (_) {
+  fdescribe('Contact Case Tab', function () {
```
Comment (swastikpareek) [inline ang/test/civicase/ContactCaseTab.spec.js]: Remove the `fdescribe` and use `describe`

**Comment by jamienovick** [comment]
Do we? Does CiviCRM core allow this?

**Comment by shitijg** [comment]
@jamienovick 
The multi-select is already existing. We need to pre fill the current contact as one of the case clients (not restrict the ability to add more clients - as it's already existing)

@reneolivo hope the core allows pre-filling one of the values by default in a multiselect field?

for eg. if I'm on Felisha Adam's contact (cases tab), and click on Add Case, Felisha Adams should be pre-filled but the user should be able to add more contacts as case clients (see screenshot)

screen shot 2018-12-05 at 14 13 01


Thanks

**Comment by reneolivo** [comment]
@jamienovick @shitijg I've been looking into this and so far what I've noticed is that when trying to populate the client it does not allow to have multiple clients when creating the case for the first time.

It seems that using PHP we can add an arbitrary contact as one of the options. Not sure how we can accomplish that since I was able to do it by modifying the code directly, but maybe there's a hook that we can use from CiviCase without modifying core. Should I look into this?

**Comment by reneolivo** [comment]
@jamienovick @swastikpareek after talking to @shitijg we agreed that we'll implement this behaviour in a future ticket after fixing all remaining issues since this is not the default behaviour from CiviCRM and it's not a priority.

**Comment by shitijg** [comment]
thanks for your time @reneolivo - yep - as @jamienovick said - we can stick to the MVP functionality for now and carry on with the other issues. 

@reneolivo let's create a new ticket and add these details on there so this is not lost. Thanks.

**Pull request merged**